bsip-0044.md

```diff
@@ -0,0 +1,173 @@
+BSIP: 0044\
+Title: Hashed Time-Locked Contract\
+Authors: Ryan R. Fox, John M. Jones, taconator\
+Status: Draft\
+Type: Protocol\
+Created: 2018-08-22\
+Discussion: TBD
+
+# **Abstract**
+
+This BSIP describes an implementation of a Hashed Time-Locked Contract (HTLC) operation.
+
+# **Motivation**
+
+The ability to securely hold tokenized assets within a hashed time-locked contract on the BitShares blockchain is a desirable feature that could be used by many persons, services, and businesses to mitigate risks between participants during asset transfer. HTLC implement conditional transfers, whereby a designated party (the "recipient") will reveal the preimage of a hash in order to execute the asset transfers from a second party (the "depositor"), else after time lock expiry "depositor" may retrieve their assets. No third-party escrow agent is required, rather the HTLC operation enforces conditions, evaluations and transfers through the BitShares consensus protocol.
+
+# **Rational**
+
+## **Elements of a Hashed Time-Locked Contract (HTLC)**
+
+An HTLC is defined to have the following components:
+
+* Parties to the HTLC
+
+    * The depositor
+
+    * The recipient
+
+* Escrow Asset
```

Comment:
I'd avoid the term "Escrow" here, to avoid confusion with #76 .

Reply:
Consider the following replacements for "Escrow Asset":
- "Guarantee"
- "Collateral"
- "Deposit Asset"
- "Secured Asset"
- "Bonded Asset"
(I listed them in my preferred order)

Reply:
Open to whatever the community thinks, but here are my opinions:
* I like escrow, as that is what it is. 
* Guarantee makes me think of some kind of written negotiated piece of paper that is distinct/abstract from the product being held. You can argue that we are doing exactly that, but I saying it just puts a different idea in my mind.
* Collateral is a good one. It would be my second choice.
* Asset is a word that is losing its meaning because of the many ways it is applied.

Reply:
I'd prefer one of "Secured/Bonded/Deposit Asset" (in that order).
Collateral has potential for confusion, just like Escrow.
Guarantee is too general IMO.

Reply:
I'm going to change it to "Secured Asset" in the next draft. Let's see how it reads throughout.